File: go/internal/scoring/testdata/structural-v1/go/state_cosmetic_before.go
package calibration

type CosmeticBefore struct{ total, updates int }

func (c *CosmeticBefore) Advance(delta int) int {
	if delta < 0 {
		return c.total
	}
	c.total += delta
	if c.total > 100 {
		c.total = 100
	}
	c.updates++
	return c.total
}

File: go/internal/scoring/testdata/structural-v1/typescript/state_cosmetic_before.ts
export class CosmeticBefore {
  private total = 0;
  private updates = 0;

  advance(delta: number): number {
    if (delta < 0) return this.total;
    this.total += delta;
    if (this.total > 100) this.total = 100;
    this.updates++;
    return this.total;
  }
}
